client_test.go

```diff
@@ -168,13 +169,17 @@ func TestClientNewSession(t *testing.T) {
 
 	client, err := New(netConn)
 	require.NoError(t, err)
-	session, err := client.NewSession(SessionIncomingWindow(incomingWindow), SessionOutgoingWindow(outgoingWindow))
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
Would a larger timeout slow down the test unnecessarily? I'm wondering whether an otherwise successful test could be flaky because it sometimes takes longer to create a session.

Reply:
For these tests they run against an in-memory fake so they should be super-fast.  I'd say that if they did start to time out it would be something to investigate.